sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -143,10 +143,21 @@ class ProjectOverviewView extends VerticalLayout{
         subscriptionCheckBox.setVisible(false)
         enableWhenProjectIsSelected(subscriptionCheckBox)
         subscriptionCheckBox.setValue(false)
+        viewModel.addPropertyChangeListener("selectedProject", {
+            Optional.ofNullable(it.newValue as ProjectSummary).ifPresent({
+                subscriptionCheckBox.value = it.hasSubscription
+            })
+        })
         subscriptionCheckBox.addValueChangeListener(event -> {
-            //Only Subscribe if checkbox is checked
-            if (subscriptionCheckBox.value && viewModel.selectedProject) {
-                subscribeToProject(viewModel.selectedProject.code)
+            if (event.oldValue == event.value) {
+                return // just to be sure
+            }
+            ProjectSummary selectedProject = viewModel.selectedProject
+            if (event.value && selectedProject) {
+                if (event.value != selectedProject.hasSubscription) {
```

Comment:
are you comparing two booleans with equals? It feels very wrong to me. Did you do it to make the statement shorter?
How about making this check in a method called like what you check here. Because for me that is not so apparent. 🙈

Reply:
I have to check for equality because I want to know if a change was requested.
The other possibility would be to check for (event.value && !selectedProject.hasSubscription) but that would only give me the information that a subscription is requested and would not give me the information that the selection changes the value.